platform_data/defs.bzl

```diff
@@ -0,0 +1,84 @@
+"""
+The platform_data rule can be used to change the target platform of a target,
+and then depend on that elsewhere in the build tree. For example:
+
+cc_binary(name = "foo")
+
+platform_data(
+    name = "foo_embedded",
+    target = ":foo",
+    platform = "//my/new:platform",
+)
+
+py_binary(
+    name = "flasher",
+    srcs = ...,
+    data = [
+        ":foo_embedded",
+    ],
+)
+
+Regardless of what platform the top-level :flasher binary is built for,
+the :foo_embedded target will be built for //my/new:platform."""
```

Comment:
This only adds default providers. Let's note that if you depend on `:foo_embedded` it's not exactly the same as depending on `:foo`, since it won't forward all the same providers. And that we can extend this to add some common providers as needed.

Reply:
Thanks! I updated the doc string with this info.